src/installer.ts

```diff
@@ -77,20 +80,57 @@ async function acquireGo(version: string): Promise<string> {
     throw new Error('Temp directory not set');
   }
 
-  if (osPlat == 'win32') {
+  // gotip is always .tar.gz.
+  if (osPlat == 'win32' && !gotip) {
     extPath = await tc.extractZip(downloadPath);
   } else {
     extPath = await tc.extractTar(downloadPath);
   }
 
+  let toolRoot = extPath;
+  if (!gotip) {
+    toolRoot = path.join(toolRoot, 'go');
+    version = normalizeVersion(version);
+  }
+
+  //
+  // Build gotip from source.
+  //
+  if (gotip) {
+    // Version embed in the executable. Required for build.
+    let versionFile = path.join(toolRoot, 'VERSION');
+    fs.writeFileSync(versionFile, `gotip-${version}`);
+
+    const options = {
+      cwd: path.join(toolRoot, 'src'),
+      env: {
+        ...process.env,
+        // Embed in the executable, set it explicitly for reproducible builds.
+        'GOROOT_FINAL': '/go',
```

Comment:
This path seems to be platform specific. Is this needed? I've never personally had to set this.

Reply:
Is `toolRoot` fine here? Since presumably that's a path for the host system.

Reply:
Yeah, should remove the leading slash. We export `GOROOT` variable, so there is no reason to embed potentially different `GOROOT_FINAL` value on each build.

Reply:
Does setup-go always export GOROOT? This would explain the trouble I had getting tip setup. Typically, you don't need to set GOROOT at all, since the go toolchain will infer it. I'd be best to leave it unset, but that's another issue.

Reply:
>Does setup-go always export GOROOT?

It does. Not sure if it’s best left unset, especially in the CI environment.